qatzip/params.go

```diff
@@ -12,8 +12,9 @@ type InputBufferMode int
 const (
 	DefaultCompression        = 1
 	MinBufferLength           = 128 * 1024
+	MaxBufferLength           = 2 * 1024 * 1024 * 1024
```

Comment:
I think we should make this configurable for the case where Go increases the slice limitation and if the user is memory constrained and wants a smaller maximum. Just check that Min <= Max.